modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/context/ChartStateContext.js

```diff
@@ -169,7 +169,7 @@ function reducer(state, action) {
 
 	switch (action.type) {
 		case ADD_DATA_SET_ITEMS:
-			nextState = action.payload.keys.reduce((state, key) => {
+			nextState = [...action.payload.keys].reduce((state, key) => {
```

Comment:
> there any difference between set-add vs array-push?

Not for `n = 2` like you have here. They are both `O(1)` amortized. In practice, a set uses a hashing algorithm to decide where to store the item whereas an array just uses an index. That means that what the "constant time" (the `1`) _actually is_ in real wall-clock time will be different. But both will be so immeasurably close to zero that it simply doesn't matter (especially give that `n = 2` like I said). So, your choice should be driven by access pattern (picking the data structure that "fits" best) and not because of actual performance difference (which would be impossible to measure in this case).